File: src/linkedlist/lc_234_palindrome_linked_list/PalindromeLinkedList.java
class PalindromeLinkedList {
    public boolean isPalindrome(ListNode head) {
        List<Integer> list = new ArrayList<>();
        ListNode temp = head;
        while (temp != null) {
            list.add(temp.val);
            temp = temp.next;
        }

        int l, r;
        l = 0;
        r = list.size() - 1;
        while (l <= r){
            if (list.get(l) != list.get(r))
                return false;
            l++;
            r--;
        }
        return true;
    }
}

File: src/linkedlist/lc_234_palindrome_linked_list/palindrome_linked_list.py
from listNode import ListNode
class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        def reverse(head):
            t = head
            t1, t2 = None, None
            while t:
                t1 = t
                t = t.next
                t1.next = t2
                t2 = t1
            return t1
        
        if not head.next:
            return True
        slow = head
        fast = head.next
        while fast and fast.next:
            slow = slow.next
            fast = fast.next
            if fast:
                fast = fast.next
        
        temp = slow.next
        slow.next = None
        rev = reverse(temp)
        t1, t2 = head, rev
        while t1 and t2:
            if t1.val != t2.val:
                return False
            t1 = t1.next
            t2 = t2.next
        return True